src/gpu/mod.rs

```diff
@@ -37,3 +37,26 @@ use ocl::Device;
 lazy_static::lazy_static! {
     pub static ref GPU_NVIDIA_DEVICES: Vec<Device> = get_devices(GPU_NVIDIA_PLATFORM_NAME).unwrap_or_default();
 }
+
+#[derive(Debug)]
+pub struct LockedFile(File);
+
+use std::fs::File;
+
+use std::io;
+pub const LOCK_NAME: &str = "/tmp/bellman.lock";
+pub fn lock() -> io::Result<LockedFile> {
+    let file = File::create(LOCK_NAME)?;
+
+    #[cfg(feature = "gpu")]
```

Comment:
this shouldn't do anything if we don't use the gpu, so that file creation should be part of the cfg block

Reply:
I've pushed a new commit with a hopefully cleaner approach.